server/src/test/java/org/elasticsearch/cluster/coordination/FollowersCheckerTests.java

```diff
@@ -0,0 +1,440 @@
+            if (deterministicTaskQueue.hasRunnableTasks()) {
+                deterministicTaskQueue.runRandomTask(random());
+            } else {
+                deterministicTaskQueue.advanceTime();
+            }
+        }
+        assertThat(checkedNodes, containsInAnyOrder(otherNode1, otherNode2));
+        assertThat(followersChecker.getFaultyNodes(), empty());
+
+        checkedNodes.clear();
+        checkCount.set(0);
+        followersChecker.setCurrentNodes(discoveryNodesHolder[0]
+            = DiscoveryNodes.builder(discoveryNodesHolder[0]).remove(otherNode1).build());
+        while (checkCount.get() < 10) {
+            if (deterministicTaskQueue.hasRunnableTasks()) {
+                deterministicTaskQueue.runRandomTask(random());
+            } else {
+                deterministicTaskQueue.advanceTime();
+            }
+        }
+        assertThat(checkedNodes, contains(otherNode2));
+        assertThat(followersChecker.getFaultyNodes(), empty());
+
+        checkedNodes.clear();
+        followersChecker.setCurrentNodes(discoveryNodesHolder[0] = DiscoveryNodes.EMPTY_NODES);
+        deterministicTaskQueue.runAllTasks(random());
+        assertThat(checkedNodes, empty());
+    }
+
+    public void testFailsNodeThatDoesNotRespond() {
+        final Builder settingsBuilder = Settings.builder();
+        if (randomBoolean()) {
+            settingsBuilder.put(FOLLOWER_CHECK_RETRY_COUNT_SETTING.getKey(), randomIntBetween(1, 10));
+        }
+        if (randomBoolean()) {
+            settingsBuilder.put(FOLLOWER_CHECK_INTERVAL_SETTING.getKey(), randomIntBetween(100, 100000) + "ms");
+        }
+        if (randomBoolean()) {
+            settingsBuilder.put(FOLLOWER_CHECK_TIMEOUT_SETTING.getKey(), randomIntBetween(1, 100000) + "ms");
+        }
+        final Settings settings = settingsBuilder.build();
+
+        testBehaviourOfFailingNode(settings, () -> null,
+            (FOLLOWER_CHECK_RETRY_COUNT_SETTING.get(settings) - 1) * FOLLOWER_CHECK_INTERVAL_SETTING.get(settings).millis()
+                + FOLLOWER_CHECK_RETRY_COUNT_SETTING.get(settings) * FOLLOWER_CHECK_TIMEOUT_SETTING.get(settings).millis());
+    }
+
+    public void testFailsNodeThatRejectsCheck() {
+        final Builder settingsBuilder = Settings.builder();
+        if (randomBoolean()) {
+            settingsBuilder.put(FOLLOWER_CHECK_RETRY_COUNT_SETTING.getKey(), randomIntBetween(1, 10));
+        }
+        if (randomBoolean()) {
+            settingsBuilder.put(FOLLOWER_CHECK_INTERVAL_SETTING.getKey(), randomIntBetween(100, 100000) + "ms");
+        }
+        final Settings settings = settingsBuilder.build();
+
+        testBehaviourOfFailingNode(settings, () -> {
+                throw new ElasticsearchException("simulated exception");
+            },
+            (FOLLOWER_CHECK_RETRY_COUNT_SETTING.get(settings) - 1) * FOLLOWER_CHECK_INTERVAL_SETTING.get(settings).millis());
+    }
+
+    public void testResetsFailureCounterOnSuccess() {
```

Comment:
Reset instead of Resets?

Reply:
The intention here is "the follower checker resets its failure counter on success" for which "Resets" is right, but I renamed it in e5a39bd1c119135079cd1ccb70093f827516a00c to something else.